packages/crypto/src/transactions/verifier.ts

```diff
@@ -26,28 +26,26 @@ export class Verifier {
             return false;
         }
 
-        return Hash.verify(
-            Transaction.getHash(transaction, { excludeSecondSignature: true }),
-            secondSignature,
-            publicKey,
-        );
+        const hash = Transaction.getHash(transaction, { excludeSecondSignature: true });
+        if (transaction.version === 2) {
```

Comment:
Those `if v1 else v2` repeat a few times now for signing and verifying. I think there should be a `Hash.verify` and `Signer.sign` method which decides which verification to use under the hood based on transaction version but I will change that in another PR that has been waiting for AIP18.